### PR TITLE
fixes #4115 - added missing LICENSE file

### PR DESCRIPTION
**Pull request by lzap**
:-)

### CODE REVIEW

**Comment by dLobatog** [comment]
Tests surprisingly pass! :traffic_light: !

**Comment by domcleal** [comment]
Okay... but... :)

We have some non-GPLv3+ content in Foreman that's been included in the codebase, or vendored.  In a related discussion (about Junos), spot says that it is "very good form to include a copy of the [..] license terms in [..] LICENSE" and "bonus points if you specify what files are under the [..] license".  The spec file license field should also include them, e.g. "GPLv3+ and BSD".

I think we should ensure we list the full and correct licences for everything we ship, and provide the appropriate terms within this LICENSE file.

At a glance, I see:
- config/locales/ is MIT-licensed
- lib/tasks/convert.rake is BSD 3-clause licensed
- extras/noVNC/websockify is LGPL v3 licensed
- public/blank.html is MIT licensed (wtf)
- vendor/ I've not looked into, but probably a mixture

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+  If the disclaimer of warranty and limitation of liability provided
+above cannot be given local legal effect according to their terms,
+reviewing courts shall apply local law that most closely approximates
+an absolute waiver of all civil liability in connection with the
+Program, unless a warranty or assumption of liability accompanies a
+copy of the Program in return for a fee.
+
+                     END OF TERMS AND CONDITIONS
+
+            How to Apply These Terms to Your New Programs
```
Comment (domcleal) [inline LICENSE]: I've always been unclear on whether this section should be included, so have always left it out.  Do you know?

Comment (lzap) [inline LICENSE]: Licenses reviewed. That was fun.

IANAL but I removed everything bellow "end of terms".

LZ

On Mon, Jan 20, 2014 at 08:35:46AM -0800, Dominic Cleal wrote:

> > +PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER PROGRAMS),
> > +EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF
> > +SUCH DAMAGES.
> > +
> > -  17. Interpretation of Sections 15 and 16.
> >   +
> > -  If the disclaimer of warranty and limitation of liability provided
> >   +above cannot be given local legal effect according to their terms,
> >   +reviewing courts shall apply local law that most closely approximates
> >   +an absolute waiver of all civil liability in connection with the
> >   +Program, unless a warranty or assumption of liability accompanies a
> >   +copy of the Program in return for a fee.
> >   +
> > -                     END OF TERMS AND CONDITIONS
> >   +
> > -            How to Apply These Terms to Your New Programs
> 
> I've always been unclear on whether this section should be included, so have always left it out.  Do you know?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman/pull/1175/files#r9010010

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,1099 @@
+The Foreman repository/package is licensed as follows:
+
+* The Foreman itself is licensed under GNU GPL v3 or newer
+* config/locales/ and public/blank.html are MIT licensed
+* lib/tasks/convert.rake is under Revised BSD license
+* extras/noVNC/websockify is LGPL v3 licensed
+* vendor/assets/fonts/ is ASL 2.0 licensed
+* vendor/assets/javascripts/ace/ is under Revised BSD license
+* vendor/assets/javascripts/noVNC/ is LGPL v3, GPL v2+, MIT and BSD
+* vendor/assets/javascripts/jsdiff.js is under Original BSD license
```
Comment (domcleal) [inline LICENSE]: Are you sure that's not the revised/3-clause?

Comment (domcleal) [inline LICENSE]: Oh, diff.js.

**Comment by domcleal** [comment]
Do you think we should update the license field in the spec file?  Maybe be vague and say "GPLv3+ with exceptions" or similar, rather than listing all of them.

File: LICENSE (hunk 2 above)
Comment (lzap) [inline LICENSE]: Oh right

https://github.com/kpdecker/jsdiff/blob/master/LICENSE

fixing

On Tue, Jan 21, 2014 at 05:10:40AM -0800, Dominic Cleal wrote:

> > @@ -0,0 +1,1099 @@
> > +The Foreman repository/package is licensed as follows:
> > +
> > +\* The Foreman itself is licensed under GNU GPL v3 or newer
> > +\* config/locales/ and public/blank.html are MIT licensed
> > +\* lib/tasks/convert.rake is under Revised BSD license
> > +\* extras/noVNC/websockify is LGPL v3 licensed
> > +\* vendor/assets/fonts/ is ASL 2.0 licensed
> > +\* vendor/assets/javascripts/ace/ is under Revised BSD license
> > +\* vendor/assets/javascripts/noVNC/ is LGPL v3, GPL v2+, MIT and BSD
> > +\* vendor/assets/javascripts/jsdiff.js is under Original BSD license
> 
> Are you sure that's not the revised/3-clause?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman/pull/1175/files#r9036034

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

**Comment by lzap** [comment]
Yeah fixed, as well as the spec.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

**Comment by domcleal** [comment]
Thanks @lzap, merged as 41e9d14 for 1.4.0.

**Pull request closed without merging**